components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/AuthenticatorConfig.java

```diff
@@ -103,4 +103,20 @@ public void setAuthenticatorStateInfo(
         return idps;
     }
 
-}
\ No newline at end of file
+    /**
+     * Deep clone of AuthenticatorConfig.
+     *
+     * @return cloned AuthenticatorConfig.
+     */
+    public AuthenticatorConfig deepClone() {
```

Comment:
a better way would be to write a copy constructor.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/2543/commits/ba962eec4d670d20ff68ad1eed4ce433b5e67096